### ISSUE TITLE
Stock Quantity Worker - needs to process WAS messages

### ISSUE THREAD

**Issue by ianrkent** [oadedokun/sqs-poc#1]

**Comment by ianrkent** [comment]
Hey Olu.  just to clarify the piece of work you want me to do.

You want me to update the StockQuantity.Worker to process the WarehouseAvailableStockChangedV1 events by:
- Loading the Region doc for the SKU
- updating the appropriate Warehouse and Region entrys (or inserting new doc if not existing)
- publish the RegionStockStatusChangedV1 event if the stock status changes 
  - positive amount to 0 -> Out Of Stock
  - 0 to amount less than 10 -> Low In Stock
  - less than 10 to >= 10  -> In Stock

Anything else?

**Comment by oadedokun** [comment]
Yes, I have identified some bugs in the StockQuantityAggregate, RegionStock and WarehouseAvailableStock, StockQuantityAggregateDocDb in the StockQuantity.Domain and StockQuantity.Data namespaces. This affe
[RegionStockStatus.cs.txt](https://github.com/oadedokun/sqs-poc/files/399089/RegionStockStatus.cs.txt)
How Statuses are evaluated. See attached files:

[RegionStock.cs.txt](https://github.com/oadedokun/sqs-poc/files/399093/RegionStock.cs.txt)

[StockQuantityAggregateDocDb.cs.txt](https://github.com/oadedokun/sqs-poc/files/399076/StockQuantityAggregateDocDb.cs.txt)
[StockQuantityAggregate.cs.txt](https://github.com/oadedokun/sqs-poc/files/399077/StockQuantityAggregate.cs.txt)
[WarehouseAvailableStock.cs.txt](https://github.com/oadedokun/sqs-poc/files/399078/WarehouseAvailableStock.cs.txt)
[StockQuantityAggregate.cs.txt](https://github.com/oadedokun/sqs-poc/files/399081/StockQuantityAggregate.cs.txt)

`

**Comment by oadedokun** [comment]
[WorkerRole.cs.txt](https://github.com/oadedokun/sqs-poc/files/399256/WorkerRole.cs.txt)

**Comment by oadedokun** [comment]
[WarehouseAvailableStockChangedV1Handler.cs.txt](https://github.com/oadedokun/sqs-poc/files/399267/WarehouseAvailableStockChangedV1Handler.cs.txt)
